src/Microsoft.PowerShell.Commands.Utility/commands/utility/Sort-Object.cs

```diff
@@ -159,15 +159,8 @@ private int Heapify(List<OrderByPropertyEntry> dataToSort, OrderByPropertyCompar
                 // If we're doing a unique sort and the entry is not unique, discard the duplicate entry
                 if (Unique && !uniqueSet.Add(dataToSort[dataIndex]))
                 {
-                    discardedDuplicates++;
-                    if (dataIndex != dataToSort.Count - discardedDuplicates)
-                    {
-                        // When discarding duplicates, replace them with an item at the end of the list and
-                        // adjust our counter so that we check the item we just swapped in next
-                        dataToSort[dataIndex] = dataToSort[dataToSort.Count - discardedDuplicates];
-                        dataIndex--;
-                    }
-
+                    dataToSort.RemoveAt(dataIndex);
```

Comment:
Performing a remove operation does not seem like the right fix for performance.  @daxian-dbw please take a look.